### PR TITLE
add docfx project to generate C# API Reference

### PR DESCRIPTION
**Pull request by koepalex**
* using docfx to be able to generate full API reference
* updated .gitignore to ignore the build results of docfx
* added reference from docs/readme.md

### CODE REVIEW

File: docs/csharp-api/index.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Industrial IoT Platform C# Reference
```
Comment (marcschier) [inline docs/csharp-api/index.md:1]: Q: Why not put this under api/ folder?

File: docs/csharp-api/docfx.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{
+  "metadata": [
+    {
+      "src": [
+        {
+          "files": [
+            "**.csproj"
```
Comment (marcschier) [inline docs/csharp-api/docfx.json:7]: Isnt this better to just document projects under api/, which are public API to be consumed?

File: docs/csharp-api/index.md (hunk 1 above)
Comment (koepalex) [reply docs/csharp-api/index.md]: Moved under api folder, which is the better place

File: docs/csharp-api/docfx.json (hunk 2 above)
Comment (koepalex) [reply docs/csharp-api/docfx.json]: agreed, updated docfx config to only work with projects under api folder

**Pull request merged**